### PR TITLE
Added important parameters to SETE.py data_preprocess(): return_kmers, min_tcrs_amount

### PR DESCRIPTION
**Pull request by liel-cohen**
SETE.py:
(1) added the boolean parameter **return_kmers** to the functions **data_preprocess** and **buildFeatures**. If the return_kmers is True, the extracted kmers list is added to the list of items returned by the functions. This way, the user can use it to make sure the independent test set's processed kmers matrix only contain the kmers contained in the train set processed data, and in the same order, and apply the PCA transformation to it. The compatibility of old scripts should be retained, since the return_kmers default value is False.
(2) added the parameter **min_tcrs_amount** to the **data_preprocess** function, so the user can control the minimal amount of TCRs needed for an epitope to be included in the modeling. The compatibility of old scripts should be retained, since the min_tcrs_amount default value is 10.

### CODE REVIEW

**Comment by yanpinlu** [comment]
Hello, I have two questions to ask you. First, are the model training sets and test sets from VDJdb? What are the filtering conditions?
Second, if I want to use your model directly to predict which epitopes the TCR will bind to, where do I start? Neither of the two examples you gave seems to have saved the trained model. What is the criterion for determining which epitope TCR will bind to, the output probability of the combination of the two or something else? @liel-cohen @wonanut

**Review by wonanut — APPROVED** [review]
@liel-cohen Thank you for paying attention to this project and contributing code. The improvement suggestions you provide are reasonable. By adding parameter control, the data format is more controllable without affecting the original functions.

**Comment by wonanut** [comment]
> Hello, I have two questions to ask you. First, are the model training sets and test sets from VDJdb? What are the filtering conditions? Second, if I want to use your model directly to predict which epitopes the TCR will bind to, where do I start? Neither of the two examples you gave seems to have saved the trained model. What is the criterion for determining which epitope TCR will bind to, the output probability of the combination of the two or something else? @liel-cohen @wonanut

Thank you for your interest in this project. 

We conduct experiments on two datasets, one of which is from VDJdb and only antigen categories with enough number of TCR sequences are retained (so the remained epitopes are listed in our paper SETE: Sequence-based Ensemble learning approach for TCR Epitope binding prediction). 

For the second question, since the total amount of dataset is small, but the feature dimension is large, we only performed cross-validation on these two datasets and took the average of the cross-validation as the final result. If you want to have a predictive model that can be used directly, you need to divide the training set, test set, and validation set (optional) based on your own dataset, and then train it yourself.

Hope my answer can help you.
